packages/aws-cdk-lib/aws-cloudfront-origins/README.md

```diff
@@ -96,60 +127,58 @@ S3 bucket to allow CloudFront read-only access
-OAI and OAC can be used in conjunction with a bucket that is not public to
-require that your users access your content using CloudFront URLs and not S3 URLs directly.
-
-> Note: OAC and OAI can only be used with an regular S3 bucket origin (not a bucket configured as a website endpoint).
+> Note: If your bucket previously used OAI, you will need to manually remove the policy statement
+that gives the OAI access to your bucket after setting up OAC.
 
-The `S3BucketOrigin` class supports creating a S3 origin with OAC, OAI, and no access control (using your bucket access settings) via
-the `withOriginAccessControl()`, `withOriginAccessIdentity()`, and `withBucketDefaults()` methods respectively.
+#### Using OAC for a SSE-KMS encrypted S3 origin
 
-Setup an S3 origin with origin access control as follows:
+If the objects in the S3 bucket origin are encrypted using server-side encryption with
+AWS Key Management Service (SSE-KMS), the OAC must have permission to use the KMS key.
+Setting up an S3 origin using `S3BucketOrigin.withOriginAccessControl()` will automatically add the statement to the KMS key policy
+to give the OAC permission to use the KMS key.
+For imported keys, you will need to manually update the
+key policy yourself as CDK apps cannot modify the configuration of imported resources.
 
 ```ts
-const myBucket = new s3.Bucket(this, 'myBucket');
-new cloudfront.Distribution(this, 'myDist', {
-  defaultBehavior: {
-    origin: origins.S3BucketOrigin.withOriginAccessControl(myBucket) // Automatically creates an OAC
-  },
+const myKmsKey = new kms.Key(this, 'myKMSKey');
+const myBucket = new s3.Bucket(this, 'mySSEKMSEncryptedBucket', {
+  encryption: s3.BucketEncryption.KMS,
+  encryptionKey: kmsKey,
+  objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED,
 });
-```
-
-You can also pass in a custom S3 origin access control:
-
-```ts
-const myBucket = new s3.Bucket(this, 'myBucket');
-const oac = new cloudfront.S3OriginAccessControl(this, 'MyOAC', { signing: cloudfront.Signing.SIGV4_NO_OVERRIDE });
-const s3Origin = origins.S3BucketOrigin.withOriginAccessControl(
-  bucket, { originAccessControl: oac }
-)
 new cloudfront.Distribution(this, 'myDist', {
   defaultBehavior: {
-    origin: s3Origin
+    origin: origins.S3BucketOrigin.withOriginAccessControl(myBucket) // Automatically grants Distribution access to `myKmsKey`
   },
 });
 ```
 
-An existing S3 origin access control can be imported using the `fromOriginAccessControlId` method:
+If the S3 bucket has an `encryptionKey` defined, `S3BucketOrigin.withOriginAccessControl()`
+will update the KMS key policy by appending the following policy statement to allow CloudFront read-only access (unless otherwise specified in the `originAccessLevels` property):
 
-```ts
-const importedOAC = cloudfront.S3OriginAccessControl.fromOriginAccessControlId(this, 'myImportedOAC', {
-  originAccessControlId: 'ABC123ABC123AB',
-});
+```json
+{
+    "Statement": {
+        "Sid": "GrantOACAccessToKMS",
```

Comment:
```suggestion

```
To reflect the latest code where `Sid` is not added

Reply:
updated!